plenum/server/node.py

```diff
@@ -752,6 +766,35 @@ def setPoolParams(self):
                 self.allNodeNames, self.requiredNumberOfInstances,
                 self.minimumNodes, self.quorums))
 
+        # 3. Adjust replicas
+        self.adjustReplicas(len(self.replicas),
+                            self.requiredNumberOfInstances)
+        for r in self.replicas.values():
+            # We set new list of validators for every replica,
+            # cause cdp for every replica need to be independent
+            r.set_validators(self.allNodeNames)
+
+        # 4. Select primaries for backups
+        primaries = self.primaries_selector.select_primaries(self.viewNo)
+        if len(self.replicas) != len(primaries):
+            raise LogicError('Inconsistent number of replicas and expected primaries. '
+                             'Number of replicas={}. Expected number of primaries={}'
+                             .format(len(self.replicas), len(primaries)))
+        if self.master_replica.primaryName is not None and \
+                generateName(primaries[0], MASTER_REPLICA_INDEX) != self.master_replica.primaryName:
+            raise LogicError('Master Primary is not expected to be changed. Current master primary {}; selected {}'
+                             .format(self.master_replica.primaryName, primaries[0]))
+        for inst_id, r in self.replicas.items():
+            r.primaryName = generateName(primaries[inst_id], inst_id)
+            self.primary_selected(inst_id)
+            logger.display("{} selected primary {} for instance {} (view {})"
+                           .format(PRIMARY_SELECTION_PREFIX,
+                                   r.primaryName, inst_id, self.viewNo))
+
+        # 5. Check if master Primary is still in the list of active nodes
+        if self.master_replica.primaryName is not None and replica_name_to_node_name(self.master_replica.primaryName) not in self.allNodeNames:
+            self.master_replica.internal_bus.send(VoteForViewChange(Suspicions.NODE_COUNT_CHANGED))
```

Comment:
Probably worth another code?

Separate thought - do we really need codes? We don't seem to use them anywhere except logging, why not just using string with reason?

Reply:
Done

Reply:
I don't see any value in these codes, looks like this is just legacy,